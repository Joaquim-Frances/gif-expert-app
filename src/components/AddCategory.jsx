import React, { useState } from 'react';
import PropTypes from "prop-types";




export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleImputChange = (event) => {
        setInputValue(event.target.value);

        console.log('HandleInputChange Llamado!')
    }

    const handleSubmit = (event) =>{
        event.preventDefault();

        console.log('HandelSubmit!', inputValue );

        if( inputValue.trim().length > 2 ){ 
            setCategories( categ =>[inputValue, ...categ]);
            setInputValue('');
        }
    }

    return (
        <form className="addCategory animate__animated animate__zoomIn" onSubmit={ handleSubmit }>
            <h3>Buscar categoria: </h3>
            <p> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                placeholder="Â¿Sobre que tema quieres que busque GIFs?"
                onChange= { handleImputChange }
                className="input-category"
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
