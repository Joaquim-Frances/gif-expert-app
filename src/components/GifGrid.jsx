import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'


export const GifGrid = ({category}) => {
    
   
    const { data, loading } = useFetchGifs( category );
    

    // useEffect ( ()=> {
    //     getGif( category ).then( setImages );
    // }, [ category]);
    
    return (
        <Fragment>
        <h3> { category } </h3>
        {loading && <p> Cargando datos...</p>}
        <div className='card-grid'>
                {
                    data.map( img =>{
                        return(
                        <GifGridItem 
                        key={ img.id }
                        { ...img } />
                        )
                    })
                }
        </div>
        </Fragment>
    )
}
