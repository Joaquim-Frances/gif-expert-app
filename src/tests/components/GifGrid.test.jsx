import { shallow } from 'enzyme'
import React from 'react'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';


describe('Testing GifGrid component functions', () => {

    const category = 'Son Goku';

    test('should do a Match with the snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow( <GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
        
    })

    test('should show items when images are loaded with our useFetch', () => {

        const gif = [{
            id: 2,
            title: 'Shakyamuni',
            url: 'https://silencetimer.com'
        },{
            id: 3,
            title: 'Amitaba',
            url: 'https://silencetimer.com'
        }]
        useFetchGifs.mockReturnValue({
            data: gif,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length).toBe(gif.length);
    })
    
})
