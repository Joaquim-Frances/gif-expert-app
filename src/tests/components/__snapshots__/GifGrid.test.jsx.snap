// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component functions should do a Match with the snapshot 1`] = `
<Fragment>
  <h3>
     
    Son Goku
     
  </h3>
  <p>
     Cargando datos...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid component functions should show items when images are loaded with our useFetch 1`] = `
<Fragment>
  <h3>
     
    Son Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={2}
      key="2"
      title="Shakyamuni"
      url="https://silencetimer.com"
    />
    <GifGridItem
      id={3}
      key="3"
      title="Amitaba"
      url="https://silencetimer.com"
    />
  </div>
</Fragment>
`;
